### PR TITLE
chore: upgrade Prettier, add root config, format

### PR DESCRIPTION
**Pull request by adidahiya**
Upgrade to Prettier 3.0

Not done in this PR: migrate TSLint -> ESLint

### CODE REVIEW

**Comment by adidahiya** [comment]
Build preview link for commit "chore: upgrade Prettier, add root config, format": <strong><a href="https://output.circle-artifacts.com/output/job/7319cf61-6c4c-4a2b-8b38-3478c95ea71d/artifacts/0/packages/docs/dist/index.html">documentation</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

**Comment by adidahiya** [comment]
Build preview link for commit "Merge remote-tracking branch 'origin/develop' into ad/add-prettier-config": <strong><a href="https://output.circle-artifacts.com/output/job/318b6eb0-336f-4d19-923c-ed0c8aa82734/artifacts/0/packages/docs/dist/index.html">documentation</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

File: package.json [hunk 1]
```diff
@@ -10,6 +10,8 @@
         "build": "lerna run build",
         "clean": "lerna run clean",
         "deploy": "NODE_DEBUG=gh-pages gh-pages -d packages/docs/dist -b master",
+        "format": "prettier --write \"./**/*.{js,json,md,ts,tsx,yml}\"",
+        "format-check": "prettier --check \"./**/*.{js,json,md,ts,tsx,yml}\"",
```
Comment (adidahiya) [inline package.json:14]: this is not used in CI yet, but I imagine it being used the same as Blueprint's repo after we migrate to ESLint in a future PR.

**Review by styu — APPROVED** [review]

**Comment by adidahiya** [comment]
Build preview link for commit "Merge remote-tracking branch 'origin/develop' into ad/add-prettier-config": <strong><a href="https://output.circle-artifacts.com/output/job/2cfe5285-66d4-4ac2-a399-e8dd6c84f6fc/artifacts/0/packages/docs/dist/index.html">documentation</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

**Pull request merged**